pkg/cloud/azure/actuators/cluster/reconciler.go

```diff
@@ -17,49 +17,259 @@ limitations under the License.
+		SecurityGroupName: azure.DefaultMasterSecurityGroupName,
+	}
+	if err := s.subnetsSvc.CreateOrUpdate(s.scope.Context, subnetSpec); err != nil {
+		return errors.Wrapf(err, "failed to CreateOrUpdate master subnet for cluster %s", s.scope.Cluster.Name)
+	}
+
+	subnetSpec = &subnets.Spec{
+		Name:              azure.DefaultAgentSubnetName,
+		CIDR:              azure.DefaultAgentSubnetCIDR,
+		VNETName:          azure.DefaultVNETName,
+		SecurityGroupName: azure.DefaultMasterSecurityGroupName,
+		RouteTableName:    azure.DefaultAgentRouteTableName,
+	}
+	if err := s.subnetsSvc.CreateOrUpdate(s.scope.Context, subnetSpec); err != nil {
+		return errors.Wrapf(err, "failed to CreateOrUpdate agent subnet for cluster %s", s.scope.Cluster.Name)
+	}
+
+	internalLBSpec := &internalloadbalancers.Spec{
+		Name:       azure.DefaultInternalLBName,
+		SubnetName: azure.DefaultMasterSubnetName,
+		VNETName:   azure.DefaultVNETName,
+		IPAddress:  azure.DefaultInternalLBIPAddress,
+	}
+	if err := s.internalLBSvc.CreateOrUpdate(s.scope.Context, internalLBSpec); err != nil {
+		return errors.Wrapf(err, "failed to CreateOrUpdate master internal lb for cluster %s", s.scope.Cluster.Name)
+	}
+
+	publicIPSpec := &publicips.Spec{
+		Name: s.scope.ClusterConfig.PublicIPName,
+	}
+	if err := s.publicIPSvc.CreateOrUpdate(s.scope.Context, publicIPSpec); err != nil {
+		return errors.Wrapf(err, "failed to CreateOrUpdate master public ip for cluster %s", s.scope.Cluster.Name)
+	}
+
+	publicLBSpec := &publicloadbalancers.Spec{
+		Name:         azure.DefaultPublicLBName,
+		PublicIPName: s.scope.ClusterConfig.PublicIPName,
+	}
+	if err := s.publicLBSvc.CreateOrUpdate(s.scope.Context, publicLBSpec); err != nil {
+		return errors.Wrapf(err, "failed to CreateOrUpdate master public lb for cluster %s", s.scope.Cluster.Name)
+	}
+
 	return nil
 }
 
 // Delete reconciles all the services in pre determined order
 func (s *Reconciler) Delete() error {
-	if err := s.groupsSvc.Delete(s.scope.Context); err != nil {
-		if services.ResourceNotFound(err) {
-			return nil
+	if err := s.deleteLB(); err != nil {
+		return errors.Wrap(err, "failed to delete lb")
+	}
+
+	if err := s.deleteSubnets(); err != nil {
+		return errors.Wrap(err, "failed to delete subnets")
+	}
+
+	rtSpec := &routetables.Spec{
+		Name: azure.DefaultAgentRouteTableName,
+	}
+	if err := s.routeTableSvc.Delete(s.scope.Context, rtSpec); err != nil {
+		if !azure.ResourceNotFound(err) {
+			return errors.Wrapf(err, "failed to Delete route table %s for cluster %s", azure.DefaultAgentRouteTableName, s.scope.Cluster.Name)
+		}
+	}
+
+	if err := s.deleteNSG(); err != nil {
+		return errors.Wrap(err, "failed to delete nsg")
+	}
+
+	vnetSpec := &virtualnetworks.Spec{
+		Name: azure.DefaultVNETName,
+	}
+	if err := s.vnetSvc.Delete(s.scope.Context, vnetSpec); err != nil {
+		if !azure.ResourceNotFound(err) {
+			return errors.Wrapf(err, "failed to Delete vnet %s for cluster %s", azure.DefaultVNETName, s.scope.Cluster.Name)
```

Comment:
General comment for all error messages to expand any acronyms for clarity and lowercase everything, except in cases like proper names (which should be almost non-existent).